opentelemetry: add trace/metrics correlation section (#18914)
DataDog/documentation: content/en/opentelemetry/otel_collector_datadog_exporter.md

If you are using the Datadog Exporter to also send OpenTelemetry traces to Datadog, use the `trace_parser` operator to extract the `trace_id` from each trace and add it to the associated logs. Datadog automatically correlates the related logs and traces. See [Connect OpenTelemetry Traces and Logs][18] for more information.

> 

Reviewer: ```suggestion
```

If you are using the Datadog Exporter to also send OpenTelemetry traces to Datadog, use the `trace_parser` operator to extract the `trace_id` from each trace and add it to the associated logs. Datadog automatically correlates the related logs and traces. See [Connect OpenTelemetry Traces and Logs][18] for more information.

## Traces and metrics correlation

> Traces and metrics can be correlated using container tags obtained from attributes generated by the [`k8sattributesprocessor`][19]. For an example configuration on setting it up, please take a look at our [example file][8] in the exporter repository. To ensure that the attributes are correctly added to both the Collector and application pods, it is important to set the `OTEL_RESOURCE_ATTRIBUTES` environment variable to specify the `k8s.pod.ip` attribute pointing to the pod IP obtained from the downwards API (such as for example [here][20]). For a full example, check our [example Kubernetes chart][21].

Reviewer 2: ```suggestion
Traces and metrics can be correlated using container tags obtained from attributes generated by the [`k8sattributesprocessor`][19]. For an example setup configuration, see the [example file][8] in the exporter repository.

To ensure that the attributes are correctly added to both the Collector and application pods, set the `OTEL_RESOURCE_ATTRIBUTES` environment variable to specify the `k8s.pod.ip` attribute pointing to the pod IP obtained from the downwards API (such as this [example][20]). For a full example, see the [example Kubernetes chart][21].
```

Datadog provides out-of-the-box dashboards that you can copy and customize. To use Datadog's out-of-the-box OpenTelemetry dashboards:

1. Install the [OpenTelemetry integration][24].
2. Go to **Dashboards** > **Dashboards list** and search for `opentelemetry`:

   {{< img src="metrics/otel/dashboard.png" alt="The Dashboards list, showing two OpenTelemetry out-of-the-box dashboards: Host Metrics and Collector Metrics." style="width:80%;">}}

The **Host Metrics** dashboard is for data collected from the [host metrics receiver][25]. The **Collector Metrics** dashboard is for any other types of metrics collected, depending on which [metrics receiver][26] you choose to enable.

### Containers overview dashboard

The [Docker Stats][27] receiver generates container metrics for the OpenTelemetry Collector. The Datadog Exporter translates container metrics to their Datadog counterparts.

> Use the following configuration to enable additional attributes on the Docker Stats receiver that populate the containers overview dashboard:

Reviewer 2: ```suggestion
Use the following configuration to enable additional attributes on the Docker Stats receiver that populates the containers overview dashboard:
```

```yaml
  docker_stats:
    metrics:
      container.network.io.usage.rx_packets:
        enabled: true
      container.network.io.usage.tx_packets:
        enabled: true
      container.cpu.usage.system:
        enabled: true
      container.memory.rss:
        enabled: true
      container.blockio.io_serviced_recursive:

>         enabled: true

Reviewer 3: Can we update this to the config we are running in demo account ? There are two important  metrics `container.uptime` and `container.memory.hierarchical_memory_limit` missing

Author: Done! Thanks for checking so thoroughly.

Datadog provides out-of-the-box dashboards that you can copy and customize. To use Datadog's out-of-the-box OpenTelemetry dashboards:

1. Install the [OpenTelemetry integration][24].
2. Go to **Dashboards** > **Dashboards list** and search for `opentelemetry`:

   {{< img src="metrics/otel/dashboard.png" alt="The Dashboards list, showing two OpenTelemetry out-of-the-box dashboards: Host Metrics and Collector Metrics." style="width:80%;">}}

The **Host Metrics** dashboard is for data collected from the [host metrics receiver][25]. The **Collector Metrics** dashboard is for any other types of metrics collected, depending on which [metrics receiver][26] you choose to enable.

> ### Containers Overview dashboard

Reviewer 2: ```suggestion
### Containers overview dashboard
```

> Container metrics in Datadog are generated by the [dockerstats][27] receiver. When used in conjunction with the Datadog Exporter, it translates them to their Datadog counterparts. Besides the default attributes, some additional ones need to be enabled. To do so, use the following configuration:

Reviewer 2: ```suggestion
The [Docker Stats receiver][27] generates container metrics in Datadog. When used in conjunction with the Datadog Exporter, the receiver translates container metrics to their Datadog counterparts. 

Use the following configuration to enable additional attributes on the Docker Stats receiver that populate the containers overview dashboard:
```

Author: This isn't correct. The Docker Stats receiver doesn't generate metrics for Datadog. It generates them for the OpenTelemetry Collector, which Datadog then uses. The first sentence would be a bit incorrect here.

Author: Also, "the exporter" translates container metrics to their Datadog counterparts. Which even more underlines how bad it was for me to use the word "it" there and how it can be misunderstood :)

Reviewer 2: Thanks for the clarification. I'm still having a bit of trouble understanding the various components, so please feel free to correct my writing. What do you think about this wording?

> The Docker Stats receiver generates container metrics for the OpenTelemetry Collector, which sends the metrics to Datadog. The Datadog Exporter then translates container metrics to their Datadog counterparts.
>
> Use the following configuration to enable additional attributes on the Docker Stats receiver that populate the containers overview dashboard:

Reviewer: ```suggestion
The [Docker Stats][27] receiver generates container metrics for the OpenTelemetry Collector. The Datadog Exporter translates container metrics to their Datadog counterparts.

Use the following configuration to enable additional attributes on the Docker Stats receiver that populate the containers overview dashboard:
```
